### PR TITLE
Update Guardian to latest

### PR DESCRIPTION
**Pull request by garath**
This tweaks the SDL templates to fetch and use the "latest" Guardian version by default. 

Resolves dotnet/arcade#8075

### CODE REVIEW

**Comment by MattGal** [comment]
Do you have a run where this commit was used to demonstrate this works?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -351,5 +351,5 @@ stages:
         -TsaIterationPath $(_TsaIterationPath)
         -TsaRepositoryName "Arcade"
         -TsaCodebaseName "Arcade"
-        -TsaPublish $True
+        -TsaPublish $False
```
Comment (riarenas) [inline azure-pipelines.yml:354]: Is there a reason for this change?

Comment (garath) [reply azure-pipelines.yml:354]: Good catch. That's a mistake. I disabled publishing while I tested.

**Comment by garath** [comment]
> Do you have a run where this commit was used to demonstrate this works?

[Build 20220718.10](https://dev.azure.com/dnceng/internal/_build/results?buildId=1889138) is a successful test run. It runs just the stages needed for SDL. (The SDL step "failed" because SDL ran and found issues, which is exactly what we want it to do.)

**Review by MattGal — APPROVED** [review]

**Review by jonfortescue — APPROVED** [review]

**Review by riarenas — APPROVED** [review]

**Comment by riarenas** [comment]
@garath, this seems to have broken the official build: https://dev.azure.com/dnceng/internal/_build/results?buildId=1894173&view=logs&s=3df7d716-4c9c-5c26-9f45-11f62216640d&j=7d9eef18-6720-5c1f-4d30-89d7b76728e9

**Pull request merged**